### DISCUSSION TITLE
Welcome to Make it So - Android Discussions!

### DISCUSSION THREAD

**Discussion by marinacoelho** [FirebaseExtended/make-it-so-android#24] [Announcements]
## 👋 Welcome! 

Make it So - Android Discussions is the place where you can ask questions about the Make it So app, suggest your ideas and engage with other community members!

**Comment by LazzaAU** [comment]
Very minor adjustment I know but, I'm just following along with the Web tutorials and when I get to the end of the page I see no means of continuing to the next "part". 

Therefore my suggestion is under the heading of "whats next" where you have next is part 3 or part 4 etc, make that a clickable link to the next part of the tutorial please :) 

Otherwise thanks for the write up. I'm just learning kotlin compose so I'm hoping this will make sense to me once I go through it all :)

**Reply by AlexAbrusca** [reply]
Sorry, I just download but I cant run the program, becouse there's not the main file I suppose. I have to add it manualy in order to run it? Or I did somthing wrong in the download?

**Reply by collaxd** [reply]
Just back to pt1 and then go for next

**Reply by rodrickgardner** [reply]
I didn't find the solution at this point! Kindly assist.

**Comment by collaxd** [comment]
@marinacoelho 
I want to express my sincere gratitude for all the content provided, which has been of paramount importance to my ongoing learning in Composed. Particularly, the approach in Part 5 regarding coroutines has impressed me deeply. The way you dealt with exceptions and the ease it brought to the code were truly remarkable.

However, I've come across an intriguing question. How can I incorporate loading states in this context? In conventional approaches with listeners, it's simpler to use try-send or post, for example. However, in this scenario, handling this state seems to be more challenging. To illustrate, consider the use of the famous 'Resources' class. Do you have any ideas or suggestions on how to address this point? I appreciate in advance for your guidance.

**Comment by t-owada** [comment]
Hi @marinacoelho and the team.
I am new to android app development and right now learning the integration between Jetpack Compose and Firebase.
Your guide and this repository are very helpful for my learning. Thank you so much!

Anyway, I am stumbling in the Part 4 below.
https://firebase.blog/posts/2022/07/adding-cloud-firestore-to-jetpack-compose-app

In the article, the guide tells me to add the Listener (and removing impl) in the TasksScreen (and other modules like viewmodel) to sync the app data with the one in the Firestore.

But the latest code in the "final" directory doesn't seem to have the implementation like this.
I guess you updated the implementation of "tasks" by using Kotlin Flow instead of Listener and MutableStateMap.

But the problem for me is that TasksScreen doesn't recompose when "tasks collection" in Firestore is changed (i.e. checking a flag in a task item in TasksScreen, adding a new document in Firestore console and so on) on the execution of "final" code.

Do you have any suggestion for this? Am I missing any setting in my Firebase project?

Thanks in advance.

**Reply by Wirtix** [reply]
i have the same issue. i can't figure it out. Did u find solution?

**Comment by rodrickgardner** [comment]
Hi, your application can't seem to run because the MainActivity file is missing! How cany you help me sort this issue out?

**Reply by t-owada** [reply]
there is a [MakeItSoActivity.kt](https://github.com/FirebaseExtended/make-it-so-android/blob/main/final/app/src/main/java/com/example/makeitso/MakeItSoActivity.kt) (this is defined as the main activity in [AndroidManifest.xml](https://github.com/FirebaseExtended/make-it-so-android/blob/main/final/app/src/main/AndroidManifest.xml))


anyway, before you run the project you need to create and set up a Firebase Project (refer to below)
https://firebase.blog/posts/2022/05/adding-firebase-auth-to-jetpack-compose-app/#adding-firebase-to-your-project

**Comment by kelvs-ricafort** [comment]
kindly make a starter code in v2 of MakeItSo
